starboard/android/shared/audio_track_audio_sink_type.cc

```diff
@@ -67,6 +70,38 @@ int GetMaxFramesPerRequestForTunnelMode(int sampling_frequency_hz) {
   return (max_frames + 15) / 16 * 16;  // align to 16
 }
 
+bool HasRemoteAudioOutput() {
+#if SB_API_VERSION >= 15
+  // SbPlayerBridge::GetAudioConfigurations() reads up to 32 configurations. The
+  // limit here is to avoid infinite loop and also match
+  // SbPlayerBridge::GetAudioConfigurations().
+  const int kMaxAudioConfigurations = 32;
+  SbMediaAudioConfiguration configuration;
+  int index = 0;
+  while (index < kMaxAudioConfigurations &&
+         MediaCapabilitiesCache::GetInstance()->GetAudioConfiguration(
```

Comment:
How long can this loop take to run? `GetAudioConfiguration()` might make a lot of jni calls.

Reply:
GetAudioConfiguration() might take hundreds of ms to load all media capabilities (`MediaCapabilitiesCache::UpdateMediaCapabilities_Locked`). But this function is usually called while creating an audio sink and that's usually after the web app queries of media capabilities, so at the time HasRemoteAudioOutput() is called, MediaCapabilitiesCache should already have the  capabilities cached.

Reply:
Sounds good